src/foam/java/Authorizable.js

```diff
@@ -0,0 +1,14 @@
+/**
+ * @license
+ * Copyright 2018 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.INTERFACE({
```

Comment:
Why not just put the real interface methods here instead of a "dummy" interface that is only defined in java?

Reply:
I was following the pattern of what we do for the `Validatable` interface. I'll put them in.

Reply:
Hmm, I removed Authorizable.java and updated Authorizable.js to this:

```JavaScript
/**
 * @license
 * Copyright 2018 The FOAM Authors. All Rights Reserved.
 * http://www.apache.org/licenses/LICENSE-2.0
 */

foam.INTERFACE({
  package: 'foam.nanos.auth',
  name: 'Authorizable',

  documentation: `
    A model should implement this interface if it is authorizable, meaning some
    users are allowed to operate on (create, read, update, or delete) that
    object but others are not.
  `,

  methods: [
    {
      name: 'authorizeOnCreate',
      javaReturns: 'void',
      args: [
        { name: 'x', javaType: 'foam.core.X' }
      ],
      javaReturns: 'void',
      javaThrows: ['AuthorizationException'],
    },
    {
      name: 'authorizeOnRead',
      javaReturns: 'void',
      args: [
        { name: 'x', javaType: 'foam.core.X' }
      ],
      javaReturns: 'void',
      javaThrows: ['AuthorizationException'],
    },
    {
      name: 'authorizeOnUpdate',
      javaReturns: 'void',
      args: [
        { name: 'x', javaType: 'foam.core.X' },
        { name: 'oldObj', javaType: 'foam.core.FObject' }
      ],
      javaReturns: 'void',
      javaThrows: ['AuthorizationException'],
    },
    {
      name: 'authorizeOnDelete',
      javaReturns: 'void',
      args: [
        { name: 'x', javaType: 'foam.core.X' }
      ],
      javaReturns: 'void',
      javaThrows: ['AuthorizationException'],
    }
  ]
});
```

but it doesn't compile this way because the Java code for the interface never gets generated.

Reply:
Interfaces don't need java code, they're just method definitions. Can you paste what's generated?
Also you have javaReturns twice in each method but that shouldn't be affecting anything.

Reply:
Oh you mean the entire file isn't generated? That might be because you added it to the blacklist in genjava2.js :p

Reply:
@mcarcaso Thanks, removing from the blacklists fixed it. I didn't realize that's what they were, my bad.

Reply:
All good. You were just following an example that happened to be wrong. We should fix up Validatable to be the same  but we can do that in a separate PR.